src/gui/SubWindow.cpp

```diff
@@ -218,15 +237,30 @@ void SubWindow::resizeEvent( QResizeEvent * event )
 	// then we set the buttons and show them if needed
 	if( windowFlags() & Qt::WindowMaximizeButtonHint )
 	{
-		buttonBarWidth = buttonBarWidth + m_buttonSize.width() + buttonGap;
 		m_maximizeBtn->move( middleButtonPos );
+	// on Mac the isMaximize() function returns always false
+	// so we have to implement our own test
+#ifdef LMMS_BUILD_APPLE
+		if( isMaximizedOnMac() )
+		{
+			m_maximizeBtn->hide();
```

Comment:
Is the `->hide()` redundant?  It seems you do it on every resize regardless, no?